be/src/storage/rowset/rowset_writer.cpp

```diff
@@ -670,15 +667,12 @@ Status HorizontalRowsetWriter::_final_merge() {
             ChunkIteratorPtr itr;
             // create temporary segment files at first, then merge them and create final segment files if schema change with sorting
             if (_context.schema_change_sorting) {
-                if (_context.tablet_schema->keys_type() == KeysType::DUP_KEYS) {
+                if (_context.tablet_schema->keys_type() == KeysType::DUP_KEYS ||
+                    _context.tablet_schema->keys_type() == KeysType::PRIMARY_KEYS) {
                     itr = new_mask_merge_iterator(seg_iterators, mask_buffer.get());
                 } else if (_context.tablet_schema->keys_type() == KeysType::UNIQUE_KEYS ||
                            _context.tablet_schema->keys_type() == KeysType::AGG_KEYS) {
                     itr = new_aggregate_iterator(new_mask_merge_iterator(seg_iterators, mask_buffer.get()), false);
-                } else {
-                    return Status::NotSupported(
-                            fmt::format("final merge: schema change with sorting do not support {} type",
-                                        KeysType_Name(_context.tablet_schema->keys_type())));
                 }
```

Comment:
why remove this, it will cause trouble if we add a new key type? if it's unreachable, add a assert or CHECK instead